openprescribing/frontend/management/commands/measure_sql/vw__opioids_total_dmd.sql

```diff
@@ -0,0 +1,89 @@
+-- This SQL is checked in to the git repo at measure_sql/vw__opioids_total_dmd.sql.
+-- Do not make changes directly in BQ!  Instead, change the version in the repo and run
+--
+--     ./manage.py create_bq_measure_views
+
+#subquery to create a "simple" administration route. 
+WITH 
+simp_form AS ( 
+SELECT DISTINCT vmp, #vmp code
+                CASE 
+                    WHEN descr LIKE '%injection%' THEN 'injection' #creates "injection" as route, regardless of whether injection or infusion. this also removes injection routes, e.g.
+                    WHEN descr LIKE '%infusion%' THEN 'injection'  #s/c, i/v etc, AS often injections have many licensed routes, which would multiply the row
+                    ELSE SUBSTR(form.descr, STRPOS(form.descr,".")+1) #takes the dosage form out of the string (e.g. tablet.oral) TO leave route.
+                END AS simple_form 
+FROM dmd.ont AS ont #the coded route for dosage form, includes vmp code 
+INNER JOIN dmd.ontformroute AS form ON form.cd=ont.form #text description of route 
+  )
+
+#subquery to normalise strength to mg
+,norm_vpi AS (
+SELECT vmp, #vmp code
+       ing, #ing code
+       strnt_nmrtr_val, #numerator strength value
+       strnt_nmrtr_uom, #numerator unit of measurement
+       unit_num.descr AS unit_num, #numerator unit 
+       unit_den.descr AS unit_den, #denominator unit
+       CASE
+           WHEN unit_num.descr = 'microgram' THEN vpi.strnt_nmrtr_val / 1000 #creates miligram value from mcg value
+           WHEN unit_num.descr = 'gram' THEN vpi.strnt_nmrtr_val * 1000 #creates miligram value from gram value
+           ELSE vpi.strnt_nmrtr_val
+       END AS strnt_nmrtr_val_mg, #all listed drugs now in miligram rather than g or mcg
```

Comment:
just reviewing this now.  Not relevant here as much, as only 3 types of unit in this group of drugs, but to make this usable across different areas, this would be helpful.

Reply:
added, thanks